browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -148,13 +161,35 @@ class AdblockCnameResolveHostClient : public network::mojom::ResolveHostClient {
     // See https://crbug.com/872665
     optional_parameters->source = net::HostResolverSource::DNS;
 
+    net::DnsConfigOverrides dns_config_overrides;
+    if (context->IsTor()) {
+      // Enforce DoH for Tor
+      // TODO(darkdh): we can consider implementing
+      // "Proxy DNS when using SOCKS v5" (network.proxy.socks_remote_dns)
+      // like Firefox has so that we don't have to enforce DoH
```

Comment:
Just a heads up about https://github.com/brave/brave-browser/issues/13414 - we should be able to remove most of this machinery once we rebase onto CR89, so I'd recommend keeping the fixes here as minimal as possible!

Reply:
It seems like we should be setting this stuff when the Tor connection is created because presumably we need to always use it? Not just for this?

Reply:
Normally, when we are in Tor window, we don't need to send ANY dns queries. We route all the HTTP/HTTPS traffic to tor proxy without knowing the IP, that means we just send hostname to proxy and the Tor exit node will do the resolving.